sdk/spring/azure-spring-integration-servicebus/src/test/java/com/azure/spring/integration/servicebus/converter/ServiceBusMessageConverterTest.java

```diff
@@ -0,0 +1,224 @@
+    String springReplyChannel = "SpringReplyChannel";
+    @Test
+    public void springMessageHeaderTest() {
+        org.springframework.messaging.Message<String> springMessage;
+        IMessage serviceBusMessage;
+        org.springframework.messaging.Message<String> convertedSpringMessage;
+
+        springMessage = MessageBuilder.withPayload(payload)
+                                      //.setHeader(MessageHeaders.ID, springMessageId) // MessageHeaders.ID header is
+                                      // read-only
+                                      .setHeader(MessageHeaders.CONTENT_TYPE, springMessageContent)
+                                      .setHeader(MessageHeaders.REPLY_CHANNEL, springReplyChannel)
+                                      .build();
+        serviceBusMessage = getConverter().fromMessage(springMessage, IMessage.class);
+        assertEquals(springMessageContent, serviceBusMessage.getContentType());
+        assertEquals(springReplyChannel, serviceBusMessage.getReplyTo());
+
+        convertedSpringMessage = getConverter().toMessage(serviceBusMessage, String.class);
+        assertEquals(springMessageContent, convertedSpringMessage.getHeaders().get(MessageHeaders.CONTENT_TYPE));
+        assertEquals(springReplyChannel, convertedSpringMessage.getHeaders().get(MessageHeaders.REPLY_CHANNEL));
+    }
+
+    String azureMessageRawId = UUID.randomUUID().toString();
+    @Test
+    public void azureMessageHeaderTest() {
+        org.springframework.messaging.Message<String> springMessage;
+        IMessage serviceBusMessage;
+        org.springframework.messaging.Message<String> convertedSpringMessage;
+
+        springMessage = MessageBuilder.withPayload(payload)
+                                      .setHeader(AzureHeaders.RAW_ID, azureMessageRawId)
+                                      .build();
+        serviceBusMessage = getConverter().fromMessage(springMessage, IMessage.class);
+        assertEquals(azureMessageRawId, serviceBusMessage.getMessageId());
+
+        convertedSpringMessage = getConverter().toMessage(serviceBusMessage, String.class);
+        assertEquals(azureMessageRawId, convertedSpringMessage.getHeaders().get(AzureHeaders.RAW_ID));
+    }
+
+    String serviceBusMessageId = UUID.randomUUID().toString();
+    Duration serviceBusTimeToLive = Duration.ofSeconds(1234);
+    Instant serviceBusScheduledEnqueueTimeUtc = Instant.now();
+    String serviceBusSessionId = UUID.randomUUID().toString();
+    String serviceBusCorrelationId = UUID.randomUUID().toString();
+    String serviceBusTo = UUID.randomUUID().toString();
+    String serviceBusLabel = UUID.randomUUID().toString();
+    String serviceBusReplyToSessionId = UUID.randomUUID().toString();
+    String serviceBusPartitionKey = serviceBusSessionId; // partitionKey should same to sessionId
+    String serviceBusViaPartitionKey = UUID.randomUUID().toString();
+    @Test
+    public void serviceBusMessageHeaderTest() {
```

Comment:
How about we add test for each header? This method is hard to read.

Reply:
Add test for each header:
1. Still hard to read.
2. Will bring a lot of duplicated codes.
3. Can not check whether multiple header can work together.